floodlight/io/statsperform.py

```diff
@@ -427,3 +432,683 @@ def read_open_statsperform_tracking_data_csv(
         pitch,
     )
     return data_objects
+
+
+# ----------------------------- StatsPerform Format ---------------------------
+
+
+def _get_and_convert(dic: dict, key: Any, value_type: type, default: Any = None) -> Any:
+    """Performs dictionary get and type conversion simultaneously.
+
+    Parameters
+    ----------
+    dic: dict
+        Dictionary to be queried.
+    key: Any
+        Key to be looked up.
+    value_type: type
+        Desired output type the value should be cast into.
+    default: Any, optional
+        Return value if key is not in dic, defaults to None.
+
+    Returns
+    -------
+    value: value_type
+        Returns the value for key if key is in dic, else default. Tries type conversion
+        to `type(value) = value_type`. If type conversion fails, e.g. by trying to force
+        something like `float(None)` due to a missing dic entry, value is returned in
+        its original data type.
+    """
+    value = dic.get(key, default)
+    try:
+        value = value_type(value)
+    except TypeError:
+        pass
+    except ValueError:
+        pass
+
+    return value
+
+
+def _read_full_tracking_data_txt_single_line(
+    package: str,
+) -> Tuple[
+    int,
+    int,
+    Dict[str, Dict[str, Tuple[float, float, float]]],
+    Dict[str, Union[str, tuple]],
+]:
+    """Extracts all relevant information from a single line of StatsPerform's .txt file
+    (i.e. one frame of data).
+
+    Parameters
+    ----------
+    package: str
+        One full line from StatsPerform's .txt-file, equals one "package" according
+        to the file-format documentation.
+
+    Returns
+    -------
+    gameclock: int
+        The gameclock of the current segment in milliseconds.
+    segment: int
+        The segment ID.
+    positions: Dict[str, Dict[str, Tuple[float, float, float]]]
+        Nested dictionary that stores player position information for each team and
+        player. Has the form `positions[team][jID] = (x, y, speed)`.
```

Comment:
```suggestion
        player. Has the form ``positions[team][jID] = (x, y, speed)``.
```
Applies to other docstrings as well